app/assets/javascripts/timeout_dialog.js.erb

```diff
@@ -0,0 +1,43 @@
+(function (win, A11yDialog) {
+  var dialogElement = document.getElementById("timeout_dialog"),
+      continueButtonElement = document.getElementById("continue_session"),
+      warningTimeInMinutes =(dialogElement.getAttribute("data-warning-in-minutes")),
+      timeoutInMinutes = (dialogElement.getAttribute("data-timeout-in-minutes")),
+      claimInProgress = dialogElement.getAttribute("data-claim-in-progress"),
+      timeoutInMilliseconds = (timeoutInMinutes - warningTimeInMinutes) * 60 * 1000,
+      dialog = new A11yDialog(dialogElement);
+
+  continueButtonElement.onclick = function() { dialog.hide() };
+
+  dialog.on('hide', refreshSession);
+
+  function refreshSession() {
+    var xhr = win.XMLHttpRequest ? new XMLHttpRequest() : new ActiveXObject("Microsoft.XMLHTTP");
+    xhr.open("GET", "<%= Rails.application.routes.url_helpers.claim_refresh_session_path %>");
+    xhr.onreadystatechange = function() {
+      if (xhr.readyState > 3 && xhr.status === 200) {
+        clearInterval(win.timeoutTimer);
+        win.timeoutTimer = setInterval(showTimeoutDialog, timeoutInMilliseconds);
+      }
+    };
+    xhr.setRequestHeader("X-Requested-With", "XMLHttpRequest");
+    xhr.send();
+  }
```

Comment:
I think you can use `Rails.ajax` for this, unless you're doing something special.

Reply:
Nice, I didn't know that existed